listView.php

```diff
@@ -351,6 +400,68 @@
   </div>
 </div>
 
+<!--Below we have the code for our "modal" for the user to add a new task. The modal outputs a form that takes in various user parameters to add a task into a task list-->
+<div class="modal fade" id="addTask" tabindex="-1" aria-labelledby="addTask" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalLabel">Add Task</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <form action="" class="addTask" method="POST">
+            <div class="form-group">
+                <label for="Task Title">Task Title</label>
+                <input style="height:50px;width:400px;font-size:20pt;" type="text" class="form-control" name="addTaskTitle" aria-describedby="emailHelp">
+                <small id="taskHelp" class="form-text text-muted">Give your task a descriptive title</small>
+            </div>
+            <div class="form-group">
+                <label for="taskDescription">Task Description</label>
+                <textarea style="height:150px;width:400px;font-size:14pt;"class="form-control" id="exampleFormControlTextarea1" rows="3" name="addTaskDetail"></textarea>
+                <small id="" class="form-text text-muted">Add a description with information regarding your task</small>
+            </div>
+            <!-- Retrieve existing list of task lists and display in a dropdown list for a user to select to add a task into-->
+            <div>
+            <label for="addListID"> Choose Task List</label>
+              <br>
+              <select name = "addListID">
+                  <option value="" disabled selected>Select Task List</option>
```

Comment:
Tested this for null input(didn't select any option), and the task didn't get created. To fix this, either a `required` element can be added to the form, or the default selection can be some existing list.

Reply:
Yes, by design a task cannot be added without choosing a corresponding task list. I will look into adding a required element.